src/wallet/wallet2.cpp

```diff
@@ -1626,42 +1589,33 @@ std::string wallet2::get_subaddress_label(const cryptonote::subaddress_index& in
 //----------------------------------------------------------------------------------------------------
 void wallet2::scan_tx(const std::vector<crypto::hash> &txids)
 {
-  // Get the transactions from daemon in batches and add them to a priority queue ordered in chronological order
-  auto cmp_tx_entry = [](const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry& l, const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry& r)
-  { return l.block_height > r.block_height; };
-
-  std::priority_queue<cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry, std::vector<COMMAND_RPC_GET_TRANSACTIONS::entry>, decltype(cmp_tx_entry)> txq(cmp_tx_entry);
-  const size_t SLICE_SIZE =  100; // RESTRICTED_TRANSACTIONS_COUNT as defined in rpc/core_rpc_server.cpp, hardcoded in daemon code
-  for(size_t slice = 0; slice < txids.size(); slice += SLICE_SIZE) {
-    cryptonote::COMMAND_RPC_GET_TRANSACTIONS::request req = AUTO_VAL_INIT(req);
-    cryptonote::COMMAND_RPC_GET_TRANSACTIONS::response res = AUTO_VAL_INIT(res);
-    req.decode_as_json = false;
-    req.prune = true;
-
-    size_t ntxes = slice + SLICE_SIZE > txids.size() ? txids.size() - slice : SLICE_SIZE;
-    for (size_t i = slice; i < slice + ntxes; ++i)
-     req.txs_hashes.push_back(epee::string_tools::pod_to_hex(txids[i]));
+  // Setup a vector containing tuples of corresponding txs, tx entries, and tx hashes
+  using txq_value_type = std::tuple<cryptonote::transaction, cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry, crypto::hash>;
+  std::vector<txq_value_type> txq;
+  txq.reserve(txids.size());
```

Comment:
For the changes in `scan_tx`, as well as updating to a `get_transactions()` call in place of the old RPC code, I changed the transaction priority queue container into a vector and then sorted at the end. This provides the same behavior since processing isn't intermingled with collecting, but should speed up `scan_tx` since the whole queue is only sorted once.